be/src/exprs/runtime_filter.cpp

```diff
@@ -351,6 +359,26 @@ class RuntimePredicateWrapper {
         return Status::OK();
     }
 
+    Status change_to_bloom_filter() {
+        if (_filter_type != RuntimeFilterType::IN_OR_BLOOM_FILTER) {
+            std::stringstream msg;
+            msg << "Can not change to bloom filter because of runtime filter type is "
+                << to_string(_filter_type);
+            return Status::InvalidArgument(msg.str());
+        }
+        _is_bloomfilter = true;
+        if (_hybrid_set->size() > 0) {
+            auto it = _hybrid_set->begin();
+            while (it->has_next()) {
+                _bloomfilter_func->insert(it->get_value());
+                it->next();
+            }
+            // release in filter
+            _hybrid_set.reset(create_set(_column_return_type));
+        }
+        return Status::OK();
```

Comment:
Please update the runtime filter type in profile ~

Reply:
done